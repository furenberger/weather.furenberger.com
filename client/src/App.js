import React, {useState} from "react";

import { AppBar, Typography, Toolbar, Grid } from "@material-ui/core";

import Map from "./Map";
import Panel from "./Panel";

const App = () => {
  const [selectedState, setSelectedState] = useState({ stateName: "", id: "" });

  return (
    <React.Fragment>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6">Weather</Typography>
        </Toolbar>
      </AppBar>
      <Grid container className="grid-main">
        <Grid item xs={6} className="grid-map">
          <Map selectedState={selectedState} setSelectedState={setSelectedState} />
        </Grid>
        <Grid item xs={6}className="grid-panel">
          <Panel selectedState={selectedState} setSelectedState={setSelectedState} />
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

export default App;
