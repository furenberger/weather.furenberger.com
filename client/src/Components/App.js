import React, { useState } from "react";

import { AppBar, Typography, Toolbar, Grid } from "@material-ui/core";

import Map from "./Map";
import Panel from "./Panel";

import { getStateFlag, getWeather, isZipCode } from "../util";

const App = () => {
  const [selectedState, setSelectedState] = useState({});
  const [weather, setWeather] = useState({});
  const [zipCode, setZipCode] = useState("");

  const handleSetZipCode = async (event) => {
    const newZip = event.target.value;
    setZipCode(newZip);

    // TODO: Check if zip code is in state
    if (isZipCode(newZip)) {
      const weather = await getWeather(newZip);
      setWeather(weather);
    } else {
      setWeather({})
    }
  };

  const handleSetSelectedState = async (selectedState) => {
    const stateFlagUrl = await getStateFlag(selectedState.stateName);
    setSelectedState({
      ...selectedState,
      stateFlagUrl,
    });
  };

  return (
    <React.Fragment>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6">Weather</Typography>
        </Toolbar>
      </AppBar>
      <Grid container className="grid-main">
        <Grid item xs={6} className="grid-map">
          <Map
            selectedState={selectedState}
            setSelectedState={handleSetSelectedState}
          />
        </Grid>
        <Grid item xs={6} className="grid-panel">
          <Panel
            selectedState={selectedState}
            weather={weather}
            zipCode={zipCode}
            handleSetZipCode={handleSetZipCode}
          />
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

export default App;
