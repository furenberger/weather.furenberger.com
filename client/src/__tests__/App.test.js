import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

// import MapGL, {Source, Layer} from 'react-map-gl';

beforeEach(()=>{
  jest.clearAllMocks()
})

// jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
//   GeolocateControl: jest.fn(),
//   Map: jest.fn(() => ({
//     addControl: jest.fn(),
//     on: jest.fn(),
//     remove: jest.fn(),
//   })),
//   NavigationControl: jest.fn(),
// }));

it('renders without crashing', () => {
  const {getByText} = render(<App />);
  const title = getByText(/Weather/);
  expect(title).toBeInTheDocument();
})


