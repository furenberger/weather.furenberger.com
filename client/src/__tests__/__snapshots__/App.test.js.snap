// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test snapshot renders 1`] = `undefined`;
